doc/rst/meta/static/searchtools.js

```diff
@@ -0,0 +1,756 @@
+        if (leftTitle === rightTitle) return 0;
+        return leftTitle > rightTitle ? -1 : 1; // inverted is intentional
+      }
+      return leftScore > rightScore ? 1 : -1;
+    });
+
+    // remove duplicate search results
+    // note the reversing of results, so that in the case of duplicates, the highest-scoring entry is kept
+    let seen = new Set();
+    results = results.reverse().reduce((acc, result) => {
+      let resultStr = result.slice(0, 4).concat([result[5]]).map(v => String(v)).join(',');
+      if (!seen.has(resultStr)) {
+        acc.push(result);
+        seen.add(resultStr);
+      }
+      return acc;
+    }, []);
+
+    results = results.reverse();
+
+    // for debugging
+    //Search.lastresults = results.slice();  // a copy
+    // console.info("search results:", Search.lastresults);
+
+    // print the results
+    _displayNextItem(results, results.length, searchTerms, highlightTerms);
+  },
+
+  queryManualIndex: (query) => {
+    const filenames = Search._index.filenames;
+    const docNames = Search._index.docnames;
+    const titles = Search._index.titles;
+    const termToLocation = Search._index.index_entries_to_locations;
+    const indexEntries = Search._index.indexentries;
+    const stemmer = new Stemmer()
+    const searchTerms = new Set();
+    const excludedTerms = new Set();
+    const queryLower = query.trim().toLowerCase();
+
+    splitQuery(query.trim()).forEach((queryTerm) => {
+      const queryTermLower = queryTerm.toLowerCase();
+
+      // do not use stopwords here
+
+      // stem the word
+      let word = stemmer.stemWord(queryTermLower);
+      // select the correct list
+      if (word[0] === "-") excludedTerms.add(word.substr(1));
+      else {
+        searchTerms.add(word);
+      }
+    });
+
+    // array of [docname, title, anchor, descr, score, filename]
+    let results = [];
+
+    function saveResult(entry, locations, score) {
+      // swap around the semicolon for nested index entries
+      let displayEntry = ""
+      /*const rev = entry.split(';').reverse();
+      for (var i = 0; i < rev.length; i++) {
+        if (i == 0) {
+          displayEntry = rev[i];
+        } else {
+          displayEntry += " (";
+          displayEntry += stemmer.stemWord(rev[i]);
+          displayEntry += ")";
+        }
+      }*/
+      displayEntry = entry.replace(';', ' > ');
+
+      locations.forEach((loc) => {
+        const [docIdxStr, anchor] = loc.split('#', 2);
+        const file = parseInt(docIdxStr, 10);
+        if (0 <= file && file < docNames.length) {
+          results.push([docNames[file],
+                        displayEntry,
+                        '#' + anchor,
+                        null,
+                        score,
+                        filenames[file],
+                        "<b>" + displayEntry + "</b> in " + titles[file]
+                       ]);
+        }
+      });
+    }
+
+    if (typeof termToLocation !== 'undefined') {
```

Comment:
We might want to use
```suggestion
    if (termToLocation) {
```
Using `typeof termToLocation !== 'undefined'` or even `termToLocation !== undefined` returns `true` when `termToLocation` is `null` because `null` is considered a defined value. And that would throw an error in the line below where we call `Object.entries` on `termToLocation`.

However, my suggestion will also return false for other "non-truthy" values like `false`, `0`, and `""` all of which I think are correct to return `false` here. (But I may be wrong; if so, we should insert another check in the `if` clause like `&& termToLocation !==null)` instead.)